### PR TITLE
[HIP] clean up

### PR DESCRIPTION
**Pull request by Rombur**
This PR:
 - replaces `TODO` with `FIXME_HIP`
 - adds missing `HIP_SAFE_CALL`
 - adds asynchronous deep copy since all the necessary functions have been implemented

### CODE REVIEW

File: core/src/Kokkos_HIP_Space.hpp [hunk 1]
```diff
@@ -663,7 +647,7 @@ class HIP {
 
   ~HIP() = default;
   HIP();
-  //  explicit HIP( const int instance_id );
+  HIP(hipStream_t stream);
```
Comment (dalg24) [inline core/src/Kokkos_HIP_Space.hpp:650]: Should that constructor be `explicit`?

File: core/src/Kokkos_HIP_Space.hpp [hunk 2]
```diff
@@ -647,7 +647,7 @@ class HIP {
...
```
Comment (masterleinad) [inline core/src/Kokkos_HIP_Space.hpp:650]: Is this tested anywhere?

File: core/src/Kokkos_HIP_Space.hpp [hunk 3]
```diff
@@ -715,6 +699,8 @@ class HIP {
   static int concurrency();
   static const char* name();
 
+  hipStream_t hip_stream() const;
```
Comment (dalg24) [inline core/src/Kokkos_HIP_Space.hpp:702]: Where is that function implemented?  Same function for the new constructor.

Comment (Rombur) [reply core/src/Kokkos_HIP_Space.hpp:702]: These changes sneaked in the wrong commit.

**Pull request merged**